Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -86,21 +86,21 @@ const UPSDataAsset* UPSWorldSubsystem::GetPSDataAsset() const
 }
 
 // Set the progression system component
-void UPSWorldSubsystem::SetHUDComponent(UPSHUDComponent* MyHUDComponent)
+void UPSWorldSubsystem::SetHUDComponent(class UPSHUDComponent* MyHUDComponent)
```

Comment:
Please remove all `class` forward declarations from the `.cpp` file. Forward declarations are only useful in `.h` files to reduce dependencies and improve compile times. In `.cpp` files, the necessary headers should be included instead, which you already have included.

#### Why:
- **In `.h`**: Use `class` forwards to avoid including large headers, which speeds up compilation.
- **In `.cpp`**: You need to include the full header to use the class properly, so forward declarations are not needed.

#### Expected result
Your .h looks good now, like next:
```
// No '#include MyHUDComponent.h`, we will use `class`-forward below:
void SetHUDComponent(class UPSHUDComponent* MyHUDComponent);
```
Your .cpp has redundant `class`-forwards, it should be removed like next:
```
//.cpp
#include "UPSHUDComponent.h"  // Necessary for full class definition

// No `class`-forward, as it's already included above in .cpp
void UPSWorldSubsystem::SetHUDComponent(UPSHUDComponent* MyHUDComponent)
{
    // ...
}
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/8f3f1e6a7abaedb69b5a8783f6a04c78f962f3f4